compiler/modules/nback/build.bal

```diff
@@ -948,8 +948,210 @@ function buildCompare(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn i
                               insn.result);
         }
         _ => {
-            return err:unimplemented("compare with operands of optional type is not implemented");
+            bir:OrderType ot = insn.orderType;
+            if ot is bir:OptOrderType {
```

Comment:
This test should be in the match pattern.

Reply:
Changed the test similar to `buildEquality`